### PR TITLE
feat: Add Migrations and Seed for Captions

### LINKED ISSUE #60: Implement Database Migrations for Sprint 2
As a developer I want my data persisted in the database so I can return resources to users at a later time.

**Task**:
- [x] Add a migration to create the users table. /Zhaoyue & Alec
- [x] Add a migration to create the captions table. /Connor

**Acceptance Criteria**:
- User information is persisted in the User table in the database.
- Captions are stored in the database with a FK to posts to allow captions to be retrieved by posts.

### PR DESCRIPTION
**Pull request by zxkevin**
Added the migration for creation of captions table
and added Caption seed data

#60

### CODE REVIEW

File: packages/hackmcx-api/src/db/migrations/2022-03-08_1129_create_table_captions.cjs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+function up (knex) {
+    return knex.schema
+        .createTable('captions', function (table){
+            table.increments('id');
+            table.string('caption', 2048).notNullable();
+            table.integer('post_id').unsigned();
+            table.foreign('post_id').references('id').inTable('posts').onDelete('cascade');
+            table.float('average_rating',2,1);
```
Comment (3dxp) [inline packages/hackmcx-api/src/db/migrations/2022-03-08_1129_create_table_captions.cjs]: We just want to grab the defaults here, it should provide less compatibility issues. 

```suggestion
            table.float('acc_rating');
```

**Comment by 3dxp** [comment]
Nice work on this @zxkevin. Looks good to me. Should be good to merge after comments are addressed.

**Comment by acampb65** [comment]
Looks good to me!

**Comment by acampb65** [comment]
We probably want to add the notNullable() method to the username, first_name, and last_name columns of the users table during migration.

**Comment by acampb65** [comment]
`2022-03-09_1502.cjs` should probably be renamed to something like `2022-03-09_1502_demo_captions.cjs`

**Comment by zxkevin** [comment]
> 

I left the name fields nullable for maybe the anonymous users in the future and just to give people the option to decide if or not they want to include their name. The username is notnullable by default due to it being the primary key.

**Review by 3dxp — APPROVED** [review]

**Review by acampb65 — APPROVED** [review]
Looks good

**Pull request merged**